File: examples/fire_wizard/character.js
// Fire Wizard — glass cannon with a fan of burning shots.
// Demonstrates: arrays, for loops, building one big object from many helpers.

function makeFireball(damage, count, spread) {
  return {
    name: "Fireball",
    key: "space",
    cooldownMs: 800,
    cast: {
      kind: "projectile",
      color: "orange",
      speed: 520,
      radius: 7,
      lifetimeMs: 1500,
      count: count,
      spreadDeg: spread,
      pierce: false,
      onHit: [
        { effect: "damage", amount: damage },
        { effect: "dot", dps: 8, durationMs: 1500 }, // a small burn
      ],
    },
  };
}

function buildCharacter() {
  // Pick a colour by walking an array.
  const flameColors = ["yellow", "orange", "red"];
  let pick = flameColors[0];
  for (let i = 0; i < flameColors.length; i++) {
    if (flameColors[i] === "orange") {
      pick = flameColors[i];
    }
  }

  // A second power: a slow-burning ground area you can drop on enemies.
  const inferno = {
    name: "Inferno",
    key: "q",
    cooldownMs: 5000,
    cast: {
      kind: "area",
      color: "red",
      radius: 90,
      durationMs: 2500,
      tickIntervalMs: 250,
      onTick: [
        { effect: "dot", dps: 12, durationMs: 800 },
        { effect: "slow", factor: 0.7, durationMs: 600 },
      ],
    },
  };

  return {
    characterName: "Fire Wizard",
    color: pick,
    size: 26,
    speed: 240,
    maxHealth: 80,
    powers: [
      makeFireball(20, 1, 0),
      inferno,
    ],
  };
}

File: examples/fire_wizard/character.py
def build_character():
    # Fire Wizard — ranged caster.
    # Demonstrates: PROJECTILE (3-shot fan) + AREA (burning ground / DoT).

    fireball = {
        "name": "Fireball",
        "key": "space",
        "cooldownMs": 700,
        "cast": {
            "kind": "projectile",
            "color": "orange",
            "speed": 600,
            "radius": 8,
            "lifetimeMs": 1200,
            "count": 3,
            "spreadDeg": 22,
            "onHit": [
                {"effect": "damage", "amount": 14},
                {"effect": "dot", "dps": 12, "durationMs": 1500},
            ],
        },
    }

    firewall = {
        "name": "Firewall",
        "key": "q",
        "cooldownMs": 6000,
        "cast": {
            "kind": "area",
            "color": "red",
            "radius": 90,
            "durationMs": 4000,
            "tickIntervalMs": 300,
            "followOwner": False,
            "onTick": [{"effect": "damage", "amount": 6}],
        },
    }

    blink = {
        "name": "Phase Step",
        "key": "e",
        "cooldownMs": 4000,
        "cast": {
            "kind": "dash",
            "color": "magenta",
            "distance": 180,
            "durationMs": 200,
            "invulnerable": True,
        },
    }

    return {
        "characterName": "Fire Wizard",
        "color": "#ff6a3d",
        "size": 22,
        "speed": 230,
        "maxHealth": 90,
        "powers": [fireball, firewall, blink],
    }
